internal/db/advisory_locks.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 Canonical.
+
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+// LockID represents a unique identifier for an advisory lock in the database.
+type LockID int64
```

Comment:
this does not need to be exported as it is not used by any of the exported methods.

Reply:
As comment above, I'll add to other PR. Sorry.